### PR TITLE
First release for Li thresholding method

### PR DESCRIPTION
**Pull request by Madeleine01**

### CODE REVIEW

File: src/image/transform/mask/li.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    let num_back; /* number of background pixels at a given threshold */
+    let num_obj;  /* number of object pixels at a given threshold */
+    let old_thresh;
+    let new_thresh;
+    let mean_back; /* mean of the background pixels at a given threshold */
+    let mean_obj;  /* mean of the object pixels at a given threshold */
+    let mean;  /* mean gray-level in the image */
+    let tolerance; /* threshold tolerance */
+    let temp;
+    tolerance = 0.5;
```
Comment (targos) [inline src/image/transform/mask/li.js]: Tolerance is hard coded to 0.5 here. Would it make sense to allow an option for this value?

File: src/image/transform/mask/li.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        sum_obj = 0;
+        num_obj = 0;
+        for (let ih = threshold + 1; ih < histogram.length; ih++) {
+            sum_obj += ih * histogram[ih];
+            num_obj += histogram[ih];
+        }
+        mean_obj = (num_obj === 0 ? 0.0 : (sum_obj / num_obj));
+        temp = (mean_back - mean_obj) / (Math.log(mean_back) - Math.log(mean_obj));
+
+        if (temp < -2.220446049250313E-16) {
```
Comment (targos) [inline src/image/transform/mask/li.js]: You don't have to copy the value, it is available with `Number.EPSILON`

File: src/image/transform/mask/li.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        mean += ih * histogram[ih];
+    }
+
+    mean /= total;
+    /* Initial estimate */
+    new_thresh = mean;
+
+    do {
+        old_thresh = new_thresh;
+        threshold = old_thresh + 0.5;	/* range */
```
Comment (targos) [inline src/image/transform/mask/li.js]: the java code is casting the value to `(int)`. It very important to do the same in JavaScript if we want the same result. This is possible with the bitwise `or` operator: `threshold = (old_thresh + 0.5) | 0`

**Comment by targos** [comment]
landed in d288d233a6d1fd6e830bef31b55a8e4ab8ef0788...ae3233ea3b327a513a50d7c3a05c07e9fe8540d2

File: src/image/transform/mask/li.js (hunk 1 above)
Comment (cballesteros) [inline src/image/transform/mask/li.js]: In the ImageJ implementation the tolerance was pre-defined, but create a parameter to this value it's not bad.

**Pull request closed without merging**